Recipe management server

portmil opened the pull request:
Request handling for fetching all recipes (of a specific user), fetching a single recipe, deleting a recipe, creating a category, fetching all categories (of a specific user), deleting a category, adding recipes to a category, and deleting recipes from a category.

portmil committed:
Add requests for recipe management

Adds request handling for fetching all recipes (of a specific user), fetching a single recipe, and deleting a recipe.

portmil committed:
Implement server side category management

Adds request handling for creating a category, fetching all categories (of a specific user), deleting a category, adding a recipe to a category, and deleting a recipe from a catecory.

portmil committed:
Update REST Client test requests

viivi-uhari commented on server/controllers/recipes.js:
```diff
@@ -1,33 +1,98 @@
+})
+
+
+recipesRouter.post('/', async (request, response, next) => {
```
Add each recipe to the "All" category.

viivi-uhari commented on server/controllers/recipes.js:
```diff
@@ -1,33 +1,98 @@
+    link,
+    description,
+    rating: rating || 0,
+    lastMakingDate,
```
Could there be a default value for `lastMakingDate`?

viivi-uhari commented on server/controllers/categories.js:
```diff
@@ -0,0 +1,141 @@
+})
+
+
+categoriesRouter.delete('/:id', async (request, response, next) => {
```
Deleting the "All" and other possible default categories should not be possible (as a note for this PR or for the future).

viivi-uhari reviewed (commented):
Looks good and some of the changes that I commented can be made to this PR or to another one in the future. To summarize, let's remember to include the "All" category. It could also be thought about how and when the other default categories will be created for a user.

portmil committed:
Add uniqueValidator to recipes

portmil committed:
Implement default categories

Default categories (for now, only 'All' and 'Breakfast') are automatically created on registration. New recipes are added to the 'All' category when creating them. User cannot remove recipes from 'All'. User cannot delete default categories.

viivi-uhari commented on server/controllers/users.js:
```diff
@@ -23,6 +24,23 @@ usersRouter.post('/', async (request, response, next) => {
+    const breakfast = new Category({
+      name: 'Breakfast',
+      icon: 'breakfast_icon',
+    })
```
When the amount of default categories grows big, they could probably be moved somewhere as constants.

viivi-uhari commented:
LGTM ☺️
